src/main/java/yanagishima/servlet/CommentServlet.java

```diff
@@ -1,165 +1,150 @@
+            if (like.isEmpty()) {
+                requireNonNull(content, "content is null");
                 Optional<Comment> comment = db.singleComment("datasource = ? and engine = ? and query_id = ?",
-                                                                           context.getDatasource(), context.getEngine(), context.getQueryId());
+                                                                           datasource, engine, queryId);
                 if (comment.isPresent()) {
                     String updateSql = format("UPDATE comment SET user = '%s', content = '%s', update_time_string = '%s' "
                                               + "WHERE datasource = '%s' and engine = '%s' and query_id = '%s'",
-                                              user, context.getContent(), updateTimeString, context.getDatasource(), context.getEngine(), context.getQueryId());
+                                              user, content, updateTimeString, datasource, engine, queryId);
                     db.updateBySQL(updateSql);
-                    responseBody.put("content", context.getContent());
+                    responseBody.put("content", content);
                     responseBody.put("likeCount", comment.get().getLikeCount());
                 } else {
-                    db.insert(Comment.class, value("datasource", context.getDatasource()),
-                       value("engine", context.getEngine()),
-                       value("query_id", context.getQueryId()),
+                    db.insert(Comment.class, value("datasource", datasource),
+                       value("engine", engine),
+                       value("query_id", queryId),
                        value("user", user),
-                       value("content", context.getContent()),
+                       value("content", content),
                        value("like_count", 0),
                        value("update_time_string", updateTimeString));
-                    responseBody.put("content", context.getContent());
+                    responseBody.put("content", content);
                     responseBody.put("likeCount", 0);
                 }
             } else {
                 Comment likedComment = db.singleComment("datasource = ? and engine = ? and query_id = ?",
-                                                                      context.getDatasource(), context.getEngine(), context.getQueryId()).get();
-                int likeCount = 0;
-                if (context.getLike().length() == 0) {
-                    likeCount = likedComment.getLikeCount() + 1;
-                } else {
-                    likeCount = likedComment.getLikeCount() + Integer.parseInt(context.getLike());
-                }
+                                                                      datasource, engine, queryId).get();
+                int likeCount = likedComment.getLikeCount() + like.orElse(1);
```

Comment:
It is in the `like.isEmpty else` block, like.orElse seems not to be necessary. 
Further like could be declared as `@Nullable @RequestParam int like` 🤔